### PR TITLE
Add MVK climate reproducibility test

### PR DESCRIPTION
**Pull request by jhkennedy**
Motivation:
--------------

Requiring model changes to pass stringent tests before being accepted as part of E3SM’s main development branch is critical for quickly and efficiently producing a trustworthy model. Depending on their impacts on model output, code modifications can be classified into three types:

1. Technical changes that continue to produce bit-for-bit identical solutions
2. Changes that cause the model solution to differ, yet produce a statistically identical climate when averaged over a sufficiently long time
3. Changes that lead to a different model climate

Only (3) impacts model climate, and changes of this type should only be implemented within the code after an in-depth demonstration of improvement. However, distinguishing between (2) and (3) requires a comprehensive analysis of both a baseline climate and the currently produced climate.

This PR: 
-----------

MVK, a E3SM-CIME `SystemTest` has been added that will launch a multivariate climate reproducibility test based on the Kolmogorov-Smirnov test. This test will be launched using the `create_test` scripts, *however*, since these tests use high level statistics, they have additional python dependencies which
need to be installed on your system and accessible via the compute nodes (if you're on a batch machine).

A `README.md` describing the usage of this test and a `requirements.txt` file describing the dependencies has been added into the new `cime/scripts/climate_reporducibility` directory. For more detail see the [README.md](https://github.com/E3SM-Project/E3SM/tree/jhkennedy/cime/mvk/cime/scripts/climate_reproducibility). 

[B4B]

### CODE REVIEW

File: cime/scripts/lib/CIME/SystemTests/mvk.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+        # namelist specifications for each instance
+        for iinst in range(1, ninst+1):
+            with open('user_nl_cam_{:04d}'.format(iinst), 'w') as nl_atm_file:
+                nl_atm_file.write('new_random = .true.\n')
+                nl_atm_file.write('pertlim = 1.0e-10\n')
+                nl_atm_file.write('seed_custom = {}\n'.format(iinst))
+
+
+    def run_phase(self):
```
Comment (jgfouca) [inline cime/scripts/lib/CIME/SystemTests/mvk.py]: This override is unnecessary since the implementation is an exact match of the parent's.

Comment (jhkennedy) [reply cime/scripts/lib/CIME/SystemTests/mvk.py]: Ah right, yep! I forgot to clean that part up; previous iterations did stuff there. Thanks!

**Comment by jgfouca** [comment]
@jhkennedy , this PR is marked WIP, is it not ready to be merged?

**Comment by jgfouca** [comment]
@jhkennedy ?

**Comment by jhkennedy** [comment]
@jgfouca , I marked it WIP to give  @salilmahajan and @PeterCaldwell a little time to try out the test. From my perspective it's ready to merge.

**Comment by PeterCaldwell** [comment]
FYI I'm probably not going to have time to try out the test :-( Let me know if that's a huge problem.

**Comment by jhkennedy** [comment]
@jgfouca , @salilmahajan and I talked offline -- he happy with is as well, so this is ready to merge.

**Comment by jgfouca** [comment]
Skipping integration testing since there's no possible impact to nightlies.

**Pull request merged**